src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs

```diff
@@ -40,14 +40,20 @@ public WebHostBuilder()
             _configureServicesDelegates = new List<Action<IServiceCollection>>();
             _configureLoggingDelegates = new List<Action<ILoggerFactory>>();
 
+            _config = new ConfigurationBuilder()
+                .AddEnvironmentVariables(prefix: "ASPNETCORE_")
+                .Build();
+
             // This may end up storing null, but that's indistinguishable from not adding it.
             UseSetting(WebHostDefaults.EnvironmentKey, Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT")
```

Comment:
It seems strange to read ASPNETCORE_ENVIRONMENT and URLS here when we just added ASPNETCORE_ above. What if we restructured this like:
```
UseSetting(WebHostDefaults.EnvironmentKey, Environment.GetEnvironmentVariable("Hosting:Environment") ?? Environment.GetEnvironmentVariable("ASPNET_ENV"));
UseSetting(WebHostDefaults.ServerUrlsKey, Environment.GetEnvironmentVariable("ASPNETCORE_SERVER.URLS"));
UseConfiguration(new ConfigurationBuilder().AddEnvironmentVariables(prefix: "ASPNETCORE_").Build());
```